### PR TITLE
New date-time package creation, initial Calendar component refactor

### PR DESCRIPTION
**Pull request by lorejoh12**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Initial check in of the new date-time package. Including the first steps of the refactor for the Calendar component, which included making the component css-in-js. This re-work also fixes a few bugs along the way, notably with jankiness when click-drag-mouseUp on the datepicker would screw up the highlighting and prevent selection.

Still TODO for the refactor:
- unit tests
- move DatePicker into new package
- accessibility/high contrast pass
- include the year picker control

We should keep in mind that this will result in two copies of controls for a little while, so changes that go into the existing date utilities and DatePicker/Calendar controls will need to be made here as well if the bugs exist both places.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/date-time/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# [Office UI Fabric React DateTime](http://dev.microsoft.com/fabric)
+
+##### DateTime components for Office UI Fabric React
+
+These are not production ready components and __should never be used in product__. This DateTime space is useful for testing new components whose APIs might change before final release.
+
```
Comment (cliffkoh) [inline packages/date-time/README.md]: :+1:

**Comment by cliffkoh** [comment]
(My review is of limited scope and primarily focused on what I was a codeowner for).

File: packages/date-time/src/components/Calendar/Calendar.base.tsx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      showMonthPickerAsOverlay: showMonthPickerAsOverlay,
+      overlayedWithButton: overlayedWithButton,
+      showGoToToday: showGoToToday,
+      showWeekNumbers: showWeekNumbers
+    });
+
+    return (
+      <div
+        className={css(rootClass, classes.root, className, 'ms-slideDownIn10')}
+        role="application"
```
Comment (jhellesen) [inline packages/date-time/src/components/Calendar/Calendar.base.tsx]: >application [](start = 14, length = 11)

Do we really want application here?  The Calendar is composed of standard HTML components, right?

File: packages/date-time/src/components/Calendar/Calendar.base.tsx [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    } else if (this.state.isMonthPickerVisible && this._monthPicker.current) {
+      this._monthPicker.current.focus();
+    }
+  }
+
+  private _renderGoToTodayButton = (classes: IProcessedStyleSet<ICalendarStyles>) => {
+    const { showGoToToday, strings } = this.props;
+    return (
+      showGoToToday && (
+        <button
```
Comment (jhellesen) [inline packages/date-time/src/components/Calendar/Calendar.base.tsx]: Do we prefer HTML buttons in general here?

File: packages/date-time/src/components/Calendar/Calendar.types.ts [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+  monthPickerOnly?: boolean;
+
+  showMonthPickerAsOverlay?: boolean;
+
+  overlayedWithButton?: boolean;
+
+  showGoToToday?: boolean;
+
+  showWeekNumbers?: boolean;
```
Comment (jhellesen) [inline packages/date-time/src/components/Calendar/Calendar.types.ts]: Missed comments!

File: packages/date-time/src/components/Calendar/CalendarDay/CalendarDay.base.tsx [hunk 5]
```diff
@@ -0,0 +1,634 @@
...
+   * corners to be rounded depending on which date range type and where the cell is located in the current grid.
+   * If we just round all of the corners, there isn't a good overlap and we get gaps between contiguous day boxes
+   * in Edge browser.
+   *
+   */
+
+  private _getWeekCornerStyles(
+    classNames: IProcessedStyleSet<ICalendarDayStyles>,
+    weeks: IDayInfo[][],
+    dateRangeType: DateRangeType
```
Comment (jhellesen) [inline packages/date-time/src/components/Calendar/CalendarDay/CalendarDay.base.tsx]: >dateRangeType: DateRangeType [](start = 4, length = 28)

Doesn't look like you're using this.

File: packages/date-time/src/components/Calendar/Calendar.base.tsx (hunk 3 above)
Comment (lorejoh12) [reply packages/date-time/src/components/Calendar/Calendar.base.tsx]: yes and no. Originally I added these as DefaultButtons and overrode the styling, but advice from dzearing suggested we do these as standard buttons for now while the Button control gets refactored

File: packages/date-time/src/components/Calendar/CalendarDay/CalendarDay.base.tsx (hunk 5 above)
Comment (lorejoh12) [reply packages/date-time/src/components/Calendar/CalendarDay/CalendarDay.base.tsx]: true, fixing.

File: packages/date-time/src/components/Calendar/Calendar.types.ts (hunk 4 above)
Comment (lorejoh12) [reply packages/date-time/src/components/Calendar/Calendar.types.ts]: good catch, fixing.

File: packages/date-time/src/components/Calendar/Calendar.base.tsx (hunk 2 above)
Comment (lorejoh12) [reply packages/date-time/src/components/Calendar/Calendar.base.tsx]: Based on this article you found: https://www.marcozehe.de/2012/02/06/if-you-use-the-wai-aria-role-application-please-do-so-wisely/
no, we don't actually need that role. Removing.

File: packages/date-time/src/demo/GettingStartedPage.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      <div className="ms-GettingStartedPage">
+        <div className="ms-GettingStartedPage-banner">
+          <h1>Office UI Fabric React DateTime!</h1>
+          <h3>A library of reusable, generic React components</h3>
+        </div>
+
+        <h2>Overview</h2>
+
+        <p>
+          Fabric React components are built as production ready components to be used in Microsoft products, but generalized, documented,
```
Comment (dzearing) [inline packages/date-time/src/demo/GettingStartedPage.tsx]: feel free to replace this later :)

Comment (lorejoh12) [reply packages/date-time/src/demo/GettingStartedPage.tsx]: will do, next round!

**Pull request merged**